lib/Db/ImageMapper.php

```diff
@@ -108,16 +108,29 @@ public function findImagesWithoutFaces(IUser $user = null) {
 		return $images;
 	}
 
-	public function imageProcessed(Image $image, array $faces, int $duration) {
+	/**
+	 * Writes to DB that image has been processed. Previously found faces are deleted and new ones are inserted.
+	 * If there is exception, its stack trace is also updated.
+	 *
+	 * @param Image $image Image to be updated
+	 * @param FaceNew[] $faces Faces to insert
+	 * @param int $duration Processing time, in milliseconds
+	 * @param \Exception|null $e Any exception that happened during image processing
+	 */
+	public function imageProcessed(Image $image, array $faces, int $duration, \Exception $e = null) {
 		$this->db->beginTransaction();
-		$currentDateTime = new \DateTime();
 		try {
 			// Update image itself
 			//
+			$error = null;
+			if ($e != null) {
+				$error = substr($e->getMessage() . "\n" . $e->getTraceAsString(), 0, 1024);
```

Comment:
is there a better way to format exception to get both message and trace?

Reply:
I do not know.. Maybe these..

https://github.com/nextcloud/3rdparty/blob/b995ca8b8c7f69a180ad6fb49989ad3e35b1367e/pear/pear_exception/PEAR/Exception.php#L443
https://github.com/nextcloud/lookup-server/blob/10e5382215e55a9d419b51db265fed0fa11f1f6d/server/vendor/slim/slim/Slim/Handlers/Error.php#L163

However, saving the backtrace in db IMHO seems unnecessary. I see better to keep a simple integer that represents an known enumerated type, so that it is simple to consider, when you want to analyze the file again.

I mean, the backtrace seems important to me for the nextcloud log.. For that the administrator can take actions to fix it, but not for the general operation of the application..

Reply:
@matiasdelellis  I was also thinking in direction to have predictable hash here, for example `md5(some fixed part of stack trace)`🤪 (because you don't care about actual exception and stack trace, only about doing aggregates + it is easier to send it outside, as it is anonymous).

In any way, I think you are right - I was using database as a logging facility! This should end up in log and not in DB. If you are OK that `error` column stays as `text`, I would write here just `$e->getMessage()`. If you think **nothing** should exists, I can change `error` column to `bool`? I prefer keeping it as `text`, so we can have at least *something* in here (I do not think it should be exposed to user from frontend anyway).

That being said, I will add new commit that moves this to lightweight variant to log only message. Push if you are OK with it, comment if you are not:)